builder/builder-main.c

```diff
@@ -358,7 +360,7 @@ main (int    argc,
   g_assert (!opt_run);
   g_assert (!opt_show_deps);
 
-  if (g_file_query_exists (app_dir, NULL) && !directory_is_empty (app_dir_path))
+  if (!opt_finish_only && g_file_query_exists (app_dir, NULL) && !directory_is_empty (app_dir_path))
```

Comment:
In the finish-only case you should actually do a reverse check (i.e. check that the app dir exists).